### PR TITLE
(SIMP-3464) Update docs and metadata

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Updated the metadata.json
* Updated the 'strings' docs
* Updated the CHANGELOG

SIMP-3464 #close

### CODE REVIEW

File: lib/puppet/parser/functions/validate_bool_simp.rb [hunk 1]
```diff
@@ -1,24 +1,28 @@
 module Puppet::Parser::Functions
-
   newfunction(:validate_bool_simp, :doc => <<-'ENDHEREDOC') do |args|
-    Validate that all passed values are either true or false. Abort catalog
-    compilation if any value fails this check.
+    Validate that all passed values are either `true` or `false`.
```
Comment (jeefberkey) [inline lib/puppet/parser/functions/validate_bool_simp.rb]: Should this be deprecated?

File: lib/puppet/parser/functions/validate_float.rb [hunk 2]
```diff
@@ -1,19 +1,19 @@
 module Puppet::Parser::Functions
-  newfunction(:validate_float, :doc => <<-EOS
-      Validates whether or not the passed argument is a float.
+  newfunction(:validate_float, :doc => <<-EOS) do |args|
```
Comment (jeefberkey) [inline lib/puppet/parser/functions/validate_float.rb]: Should this be deprecated?

File: lib/puppet/parser/functions/validate_integer.rb [hunk 3]
```diff
@@ -1,19 +1,18 @@
 module Puppet::Parser::Functions
-  newfunction(:validate_integer, :doc => <<-EOS
-      Validates whether or not the passed argument is an integer.
+  newfunction(:validate_integer, :doc => <<-EOS) do |args|
```
Comment (jeefberkey) [inline lib/puppet/parser/functions/validate_integer.rb]: Should this be deprecated?

File: lib/puppet/parser/functions/validate_port.rb [hunk 4]
```diff
@@ -1,25 +1,28 @@
 module Puppet::Parser::Functions
-  newfunction(:validate_port, :doc => <<-EOS) do |arguments|
-      Validates whether or not the passed argument is a valid port (i.e. between 1 - 65535).
+  newfunction(:validate_port, :doc => <<-EOS) do |args|
```
Comment (jeefberkey) [inline lib/puppet/parser/functions/validate_port.rb]: Should this be deprecated? The new type alias may or may not be sufficient.

File: lib/puppet/parser/functions/validate_re_array.rb [hunk 5]
```diff
@@ -1,31 +1,29 @@
 module Puppet::Parser::Functions
   # Derived from the Puppet Labs stdlib validate_re
   newfunction(:validate_re_array, :doc => <<-'ENDHEREDOC') do |args|
-    Perform simple validation of a string, or array of strings,
+    Perform simple validation of a `String`, or `Array` of `Strings`,
```
Comment (jeefberkey) [inline lib/puppet/parser/functions/validate_re_array.rb]: Should this be deprecated?

File: lib/puppet/provider/simp_file_line/ruby.rb [hunk 6]
```diff
@@ -1,4 +1,9 @@
 Puppet::Type.type(:simp_file_line).provide(:ruby) do
+  desc <<-EOM
```
Comment (jeefberkey) [inline lib/puppet/provider/simp_file_line/ruby.rb]: What's the difference between this one and `file_line`?

**Comment by heliocentric** [comment]
Yea, the facts/providers need better confinement. Also some of the functions are rhel specific and also need confinement

**Comment by trevor-vaughan** [comment]
@jeefberkey Please read the title and reasses your review against the work actually performed.

**Comment by jeefberkey** [comment]
Irrelevant comments contained in a new ticket.
* https://simp-project.atlassian.net/browse/SIMP-3487

**Pull request merged**